process/block/metablock.go

```diff
@@ -1628,6 +1633,33 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 	return nil
 }
 
+func (mp *metaProcessor) verifyEpochStartDataForMetablock(metaBlock *block.MetaBlock) error {
+	if !metaBlock.IsStartOfEpochBlock() {
```

Comment:
This method checks if the EpochStart.LastFinalizedHeaders slice contains anything and this struct has been added to all metaBlocks. Couldn't it work to extend the metaBlock struct with the new struct and have a EndOfEpochMetaBlock?

Reply:
could be a refactoring in the future